messaging/event.cc

```diff
@@ -6,17 +6,27 @@
 #include <exception>
 #include <filesystem>
 
-#include <sys/eventfd.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
 #include <unistd.h>
 #include <poll.h>
 #include <signal.h>
 #include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#ifdef __APPLE__
+#define ppoll(fds, nfds, timeout, sigmask) poll(fds, nfds, timeout != nullptr ? (timeout->tv_sec * 1000 + timeout->tv_nsec / 1000000) : -1)
+#else
+#include <sys/eventfd.h>
+#endif 
 
 #include "cereal/messaging/event.h"
 
 void event_state_shm_mmap(std::string endpoint, std::string identifier, char **shm_mem, std::string *shm_path) {
+  #ifdef __APPLE__
+  std::cerr << "event_state_shm_mmap not supported on macOS" << std::endl;
+  assert(false);
```

Comment:
doesn't the single assert in the constructor cover it?

Reply:
it could, but thought about adding this check to prevent people from ever calling it, cause it uses the Linux-only shm fs

Reply:
let's remove this